packages/main/src/plugin/kubernetes/kubernetes-client.spec.ts

```diff
@@ -2408,3 +2409,49 @@ describe('port forward', () => {
     expect(serviceMock.deleteForward).toHaveBeenCalledWith(DUMMY_FORWARD_CONFIG);
   });
 });
+
+test('Expect deleteCronJob to not be called if there is no active connection', async () => {
+  const client = createTestClient('default');
+  const deleteCronJobMock = vi.fn();
+  vi.spyOn(client, 'checkConnection').mockResolvedValue(false);
+  makeApiClientMock.mockReturnValue({
+    deleteNamespacedCronJob: deleteCronJobMock,
+  });
+
+  await client.deleteCronJob('name');
+  expect(deleteCronJobMock).not.toBeCalled();
+});
+
+test('Expect deleteCronJob to be called if there IS an active connection', async () => {
+  const client = createTestClient('default');
+  const deleteCronJobMock = vi.fn();
+  makeApiClientMock.mockReturnValue({
+    deleteNamespacedCronJob: deleteCronJobMock,
+  });
+  vi.spyOn(client, 'checkConnection').mockResolvedValue(true);
+  await client.deleteCronJob('name');
+  expect(deleteCronJobMock).toBeCalled();
+});
+
+test('Expect readNamespacedCronJob to return the secret', async () => {
```

Comment:
'secret' shows where you copied from :)

Test needs to be fixed.

Reply:
copy pasta accident! haha. I'll update the name